torch/csrc/jit/api/object.h

```diff
@@ -102,6 +102,38 @@ struct TORCH_API Object {
     });
   }
 
+  // Used as a utility function to find a property
+  using Property = std::tuple<std::string, Method, c10::optional<Method>>;
+  const std::vector<Property> get_properties() const {
+    return c10::fmap(type()->properties(), [&](ClassType::Property prop) {
+      c10::optional<Method> setter = c10::nullopt;
+      if (prop.setter) {
+        setter = Method(_ivalue(), prop.setter);
+      }
+      return std::make_tuple(prop.name, Method(_ivalue(), prop.getter), setter);
+    });
+  }
+
+  bool has_property(const std::string& name) const {
+    auto props = get_properties();
+    for (auto prop : props) {
+      if (std::get<0>(prop) == name) {
```

Comment:
Consider returning a `struct` from `get_properties` so you can do `prop.name` here instead. That would be more readable.